resources/lib/iptvmanager.py

```diff
@@ -57,16 +59,24 @@ def send_epg(self):  # pylint: disable=no-method-argument,no-self-use
         redbull = RedBullTV()
 
         epg = defaultdict(list)
+
         for item in redbull.get_epg().get('items'):
-            epg['redbulltv'].append(dict(
-                start=datetime.strptime(item.get('start_time'), "%Y-%m-%dT%H:%M:%S.%fZ").replace(tzinfo=UTC).isoformat(),
-                stop=datetime.strptime(item.get('end_time'), "%Y-%m-%dT%H:%M:%S.%fZ").replace(tzinfo=UTC).isoformat(),
-                title=item.get('title'),
-                description=item.get('long_description'),
-                subtitle=item.get('subheading'),
-                genre='Sport',
-                image=redbull.get_image_url(item.get('id'), item.get('resources'), 'landscape'),
-                stream=url_for('play_uid', uid=item.get('id'))
-            ))
+            if (
+                    item.get('start_time')
+                    and item.get('end_time')
+                    and re.match('\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}.\d{3}Z', item.get('start_time')) # pylint: disable=anomalous-backslash-in-string
+                    and re.match('\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}.\d{3}Z', item.get('end_time'))): # pylint: disable=anomalous-backslash-in-string
```

Comment:
Why was this needed?

Reply:
I thought I'd we can have missing start/end times that we need to catch, maybe also catch incorrect/different time stamp formats. I have not seen any though. We should log this at INFO level I think.

Reply:
I am not concerned about this, obviously we need to protect against missing entries and check the routine to report this in GitHub CI whenever this would happen. But I wouldn't handle this specifically in the add-on itself just yet. This is not very efficient.

So I would let it die with an exception for conditions we don't expect to happen. Otherwise we may not know about them if they do happen and they are not "in your face". Let's work on a Status workflow that also runs this code path to ensure that it is safe to use.